google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/cloud/CloudSdkAppEngineHelper.java

```diff
@@ -143,18 +144,26 @@ public Project getProject() {
           deploy, isFlexCompat));
     } else if (targetEnvironment.isFlexible()) {
       try {
-        // Checks if the Yaml or Dockerfile exist.
-        Optional<FlexibleRuntime> runtimeOptional =
-            AppEngineProjectService.getInstance().getFlexibleRuntimeFromAppYaml(
-                deploymentConfiguration.getAppYamlPath());
+        AppEngineFlexibleFacet flexFacet =
+            AppEngineFlexibleFacet.getFacetByModuleName(
+                deploymentConfiguration.getModuleName(), project);
 
-        if (!Files.exists(Paths.get(deploymentConfiguration.getAppYamlPath()))) {
+        if (flexFacet == null
+            || !Files.exists(Paths.get(flexFacet.getConfiguration().getAppYamlPath()))) {
```

Comment:
I guess there are two different potential problems here:
1) the module name that was saved doesn't exist anymore (it was removed or renamed)
2) the module flex facet doesn't have an actual app.yaml file specified

for 1 can we check this when the user brings up the runtime config editor and unset the module if it doesn't exist anymore?